metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala

```diff
@@ -68,6 +68,80 @@ class StringActions(buffers: Buffers, trees: Trees) extends CodeAction {
                 Nil
             }
 
+            val interpolationActions = tokens.collect {
+              case token: Token.Constant.String
+                  if token.pos.toLSP.overlapsWith(range) =>
+                token
+            }.toList match {
+              case (t: Token.Constant.String) :: _ =>
+                List(interpolateAction(uri, t.pos.toLSP))
+              case _ :: (t: Token.Constant.String) :: _ =>
+                List(interpolateAction(uri, t.pos.toLSP))
+              case _ => Nil
+            }
+
+            val removeInterpolationActions = {
+              def isStart(tokenID: Int) = {
+                tokens.lift(tokenID) match {
+                  case Some(_: Token.Interpolation.Start) => true
+                  case _ => false
+                }
+              }
+
+              def isPart(tokenID: Int) = {
+                tokens.lift(tokenID) match {
+                  case Some(_: Token.Interpolation.Part) => true
+                  case _ => false
+                }
+              }
+
+              def isEnd(tokenID: Int) = {
+                tokens.lift(tokenID) match {
+                  case Some(_: Token.Interpolation.End) => true
+                  case _ => false
+                }
+              }
+
+              val removeInterpolationToken =
+                tokens.zipWithIndex
+                  .collect {
+                    case (start: Token.Interpolation.Start, i: Int) =>
+                      if (
+                        start.pos.toLSP.overlapsWith(range)
+                        && isPart(i + 1) && isEnd(i + 2)
```

Comment:
There might be more parts than one I think here. When encountering Start you could do something similar to 

https://github.com/scalameta/metals/blob/8c24e5d2b909dda88f39312c66089195b5e36e9c/metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala#L208

So it would probably make sense to use index instead of doing collect. But otherwise something like this should work:

```scala
case (start: Token.Interpolation.Start, i: Int) if tokens(i+2).is[Token.Interpolation.End] =>
   token
```
Shortest possible interpolation string would be Start, Part, End

Reply:
The motivation behind this solution is due to which part overlaps with `range` when considering the case of multiline interpolated strings: it's one of Start, Part, or End, based on if the cursor is in on of the following positions: 
```
s"<<">>"my awesome <<stri>>ng"<<">>"
```  

On non multiline strings the overlap is given on all three parts, and the collect returns three time the Start token, that's also why I do `.distinct` in the end.

Also, not sure it's needed, but the `isPart`, `isStart`, and `isEnd` functions also check if the index of the token exists before getting it and checking it for the correct token type, If not needed, `isInstanceOf[]` makes everything cleaner.

In general, can you help me providing an example when an interpolated strings without Splices can have more Parts in it?

Reply:
I think without splices it can only be max 3. Start, Part, End.

Why not then:
```scala
.flatMap{
  case (start: Token.Interpolation.Start, i: Int) if tokens(i+2).is[Token.Interpolation.End] =>
    def overlaps = List(start, tokens(i + 1), tokens(i +2) ).exists(_.pos.toLSP.overlapsWith(range))
    if( i +2 < tokens.length && overlaps) {
      val lspRange = start.pos.toLSP
      val editRange = new l.Range(lspRange.getStart, lspRange.getEnd)
      Some( removeInterpolationAction(uri, editRange))
    } else {
     None
    }
    
  case _ => None
}

```

Reply:
I've implemented the suggested changes in the latest commit
I just changed where to check for `i +2 < tokens.length`, by doing it before using `tokens(i+2)` in the following line:
```scala
 case (start: Token.Interpolation.Start, i: Int) if tokens(i+2).is[Token.Interpolation.End] =>
 ```
 That became
 ```scala
  case (start: Token.Interpolation.Start, i: Int) if ( i+2 < tokens.length && tokens(i+2).is[Token.Interpolation.End]) =>
  ```